invenio_records_permissions/permissions/base.py

```diff
@@ -6,99 +6,108 @@
 # and/or modify it under the terms of the MIT License; see LICENSE file for
 # more details.
 
+"""Base access controls."""
+
+from itertools import chain
+
 from flask import current_app
 from invenio_access import Permission
+
 from ..generators import Deny
 
+# Where can a property be used?
+#
+# |    Action   | need | excludes | query_filters |
+# |-------------|------|----------|---------------|
+# |    create   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |     list    |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |     read    |   x  |     x    |        x      |
+# |-------------|------|----------|---------------|
+# | read files  |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |    update   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |    delete   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+#
 
-class _PermissionConfig(object):
 
-    # Deny all by default
-    can_create = [Deny]
-    can_list = [Deny]
-    can_read = [Deny]
-    can_update = [Deny]
-    can_delete = [Deny]
-
-    @classmethod
-    def get_permission_list(cls, action):
-        if action == 'create':
-            return cls.can_create
-        elif action == 'list':
-            return cls.can_list
-        elif action == 'read':
-            return cls.can_read
-        elif action == 'update':
-            return cls.can_update
-        elif action == 'delete':
-            return cls.can_delete
-
-        current_app.logger.error("Unkown action {action}.".format(
-            action=action))
-        return []
+class BasePermissionPolicy(Permission):
+    """
+    BasePermissionPolicy to inherit from.
 
+    The class defines the overall policy and the instance encapsulates the
+    permissions for an *action* *over* a set of objects.
 
-# Where can a property be used?
-#
-# |    Action   | need | excludes | query_filter |
-# |-------------|------|----------|--------------|
-# |    create   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |     list    |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |     read    |   x  |     x    |       x      |
-# |-------------|------|----------|--------------|
-# | read files  |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |    update   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |    delete   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-#
+    TODO: Recognize a PermissionPolicy class in other modules instead of
+          individual factory functions to lessen the configuration burden.
+    """
 
+    # Deny all by default
+    can_list = [Deny()]
```

Comment:
I understand why putting this config here. However, since the definition is an *action* singular, its not fully simple. I'm not fully convinced, but creating a config for this is over engineering since I don't find any other reason than "readability".

Reply:
If  I understand correctly, you mean that we shouldn't have `Deny()` at all? I agree with that.